File: Sources/CountriesFlagsObjC/source/Countries/UIImage+Israel.m
//
//  UIImage+Israel.m
//  CountriesFlags
//

#import "Countries/UIImage+Israel.h"

#import "Extensions/UIBezierPath+Triangle.h"
#import "Extensions/UIBezierPath+AffineTransform.h"
#import "Extensions/CGSize+Extension.h"

@implementation UIImage (Israel)

+ (UIImage *)flagIsraelWithSize:(CGSize)size {
    UIColor* colorBlue = [UIColor colorWithRed:0.0f
                                         green:56.0f/255.0f
                                          blue:184.0f/255.0f
                                         alpha:1.0f];
    UIColor* colorWhite = [UIColor whiteColor];
    
    UIGraphicsBeginImageContextWithOptions(size, YES, 0.0f);
    
    [colorWhite setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorBlue setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 160.0f * 15.0f, size.width, size.height / 160.0f * 25.0f));
    UIRectFill(CGRectMake(0.0f, size.height / 160.0 * 120.0, size.width, size.height / 160.0f * 25.0f));
    
    [colorBlue setStroke];
    CGFloat triangleRadius = size.height / 160.0f * 50.5f;
    CGPoint triangleCenter = CGPointMake(size.width / 2.0f, size.height / 2.0f - size.height / 160.0f * 7.3f);
    UIBezierPath* path = [UIBezierPath triangleAtPoint:triangleCenter radius:triangleRadius];
    path.lineWidth = size.height / 160.0f * 5.4f;
    [path stroke];

    [path rotateToAngle:180.0 center:CGSizeMidPoint(size)];
    [path stroke];
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image != nil ? image : [UIImage new];
}

@end

File: Sources/CountriesFlags/Countries/UIImage+Israel.swift
//
//  UIImage+Israel.swift
//  CountriesFlags
//

#if canImport(UIKit)

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://en.wikipedia.org/wiki/Flag_of_Israel "wikipedia"
     
     Generates the flag of Israel.
     For more information see [wikipedia].

     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Israel.
     */
    internal static func flagIsrael(with size: CGSize) -> UIImage? {
        let colorBlue = UIColor(red: 0.0,
                                green: 56.0 / 255.0,
                                blue: 184.0 / 255.0,
                                alpha: 1.0)
        let colorWhite = UIColor.white
        let lineSize = CGSize(width: size.width, height: size.height / 160.0 * 25.0)

        UIGraphicsBeginImageContextWithOptions(size, true, 0.0)

        colorWhite.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))

        colorBlue.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0.0, y: size.height / 160.0 * 15.0), size: lineSize))
        UIRectFill(CGRect(origin: CGPoint(x: 0.0, y: size.height / 160.0 * 120.0), size: lineSize))

        colorBlue.setStroke()
        let triangleRadius = size.height / 160.0 * 50.5
        let triangleCenter = CGPoint(x: size.width / 2.0,
                                     y: size.height / 2.0 - size.height / 160.0 * 7.3)
        let path = UIBezierPath.triangle(at: triangleCenter, radius: triangleRadius)
        path.lineWidth = size.height / 160.0 * 5.4
        path.stroke()

        path.rotate(to: 180.0, around: size.midPoint)
        path.stroke()

        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()

        return image
    }
}

#endif
